clcache.py

```diff
@@ -864,7 +864,11 @@ def parseArgumentsAndInputFiles(cmdline):
                     else:
                         raise AssertionError("Unsupported argument type.")
 
-                    arguments[arg.name].append(value)
+                    # consistently use empty list for flags like /c or /doc
+                    if not arg.name in arguments:
+                        arguments[arg.name] = []
+                    if value:
+                        arguments[arg.name].append(value)
```

Comment:
But I just see that this is the wrong fix, since it converts `/X /Xab` to `{X: ['ab']}` instead of `{X: ['', 'ab']}`, so we're losing information.

The correct fix is to go for

```python
                    argumentName = cmdLineArgument[1:] # name not followed by parameter in this case
                    arguments[argumentName].append('')
```

below. Then switches and arguments without parameter value always get the '' value.